core/src/org/javarosa/xpath/expr/XPathFuncExpr.java

```diff
@@ -145,6 +145,17 @@ public Object eval(DataInstance model, EvaluationContext evalContext) {
 
         Hashtable funcHandlers = evalContext.getFunctionHandlers();
 
+
+        //check for custom handler, use this if it exists.
+        try {
+            IFunctionHandler handler = (IFunctionHandler) funcHandlers.get(name);
+            if (handler != null) {
+                return evalCustomFunction(handler, argVals, evalContext);
+            }
+        } catch(XPathArityException e){
+            // we matched the name but not the arg count.
+            // continue in case the default has the right arity.
+        }

```

Comment:
return type should be fine as far as I know. @wpride can you confirm that'll match for this?